imagetest/test_suites/packagevalidation/package_test.go

```diff
@@ -38,41 +61,74 @@ func TestStandardPrograms(t *testing.T) {
 func TestGuestPackages(t *testing.T) {
 	utils.LinuxOnly(t)
 	image, err := utils.GetMetadata(utils.Context(t), "instance", "image")
+
 	if err != nil {
 		t.Fatalf("couldn't determine image from metadata")
 	}
-	cmdPrefix := []string{"rpm", "-q", "--queryformat", "'%{NAME}\n'"}
+
+	// What command to list all packages
+	listPackagesCmd := []string{"rpm", "-qa", "--queryformat", "%{NAME}\n"}
 	if strings.Contains(image, "debian") || strings.Contains(image, "ubuntu") {
-		cmdPrefix = []string{"dpkg-query", "-W", "--showformat", "'${Package}\n'"}
+		listPackagesCmd = []string{"dpkg-query", "-W", "--showformat", "${Package}\n"}
+	}
+
+	// What packages to insure are not installed
+	excludePkgs := []string{}
+
+	// What packages to check for and excude for diffrent images
+	requiredPkgs := []string{"google-guest-agent", "google-osconfig-agent"}
+	if strings.Contains(image, "rhel") {
+		requiredPkgs = append(requiredPkgs, "google-compute-engine", "gce-disk-expand", "google-cloud-sdk")
 	}
-	packages := []string{"google-guest-agent", "google-osconfig-agent"}
-	if strings.Contains(image, "sles") {
-		packages = append(packages, "google-guest-configs") // SLES name for 'google-compute-engine' package.
-		packages = append(packages, "google-guest-oslogin")
-	} else {
-		packages = append(packages, "google-compute-engine")
-		packages = append(packages, "google-compute-engine-oslogin")
+	if strings.Contains(image, "sles") || strings.Contains(image, "suse") {
+		requiredPkgs = append(requiredPkgs, "google-guest-configs") // SLES name for 'google-compute-engine' package.
+		requiredPkgs = append(requiredPkgs, "google-guest-oslogin")
 	}
 	if strings.Contains(image, "centos-7") {
-		packages = append(packages, "epel-release")
-	}
-
-	for _, pkg := range packages {
-		args := append(cmdPrefix[1:], pkg)
-		cmd := exec.Command(cmdPrefix[0], args...)
-		stderr := &bytes.Buffer{}
-		cmd.Stderr = stderr
-		stdout := &bytes.Buffer{}
-		cmd.Stdout = stdout
-		if err := cmd.Start(); err != nil {
-			t.Errorf("error starting check command: %v", err)
-			continue
+		requiredPkgs = append(requiredPkgs, "epel-release")
+	}
+	if strings.Contains(image, "debian") {
+		requiredPkgs = append(requiredPkgs, "haveged", "net-tools")
+		requiredPkgs = append(requiredPkgs, "google-cloud-packages-archive-keyring", "isc-dhcp-client")
+		requiredPkgs = append(requiredPkgs, "google-compute-engine", "gce-disk-expand", "python3", "google-cloud-cli")
+		excludePkgs = append(excludePkgs, "cloud-initramfs-growroot")
+		if runtime.GOARCH == "amd64" {
+			requiredPkgs = append(requiredPkgs, "grub-efi-amd64-signed", "shim-signed")
 		}
-		if err := cmd.Wait(); err != nil {
-			t.Errorf("error running check command: %v %s %s", err, stdout, stderr)
-			continue
+		if runtime.GOARCH == "arm64" {
+			requiredPkgs = append(requiredPkgs, "grub-efi-arm64")
 		}
+	}
+
+	cmd := exec.Command(listPackagesCmd[0], listPackagesCmd[1:]...)
```

Comment:
lines 103-121 can be a lot less verbose, something like this works fine and is much easier to follow imo:
```
cmd := exec.Command(listPackagesCmd[0], listPackagesCmd[1:]...)
out, err := cmd.Output()
if err != nil {
	t.Errorf("Failed to execute list packages command: %v", err)
	return
}
installedPkgs := strings.Split(string(out), "\n")
```

Reply:
oh yeah that's way better